tests/python/relax/distributed/test_distributed_transform_propagate_sharding.py

```diff
@@ -68,13 +68,119 @@ def foo(
             x, weight1, out_dtype="void"
         )
         lv1: R.DTensor((128, 128), "float32", "mesh[0]", "S[1]") = R.nn.gelu(lv0)
-        lv2: R.DTensor((128, 128), "float32", "mesh[0]", "S[1]") = lv1
         lv3: R.DTensor((128, 128), "float32", "mesh[0]", "R") = R.matmul(
-            lv2, weight2, out_dtype="void"
+            lv1, weight2, out_dtype="void"
         )
         return lv3
 
 
+@I.ir_module
```

Comment:
Nitpick: Parsing of TVMScript should be done within the test body (if used for a single test case), or within a helper function called by the body (if used by multiple test cases).  If the TVMScript is at module-scope, parsing occurs when the module is imported.  This causes the entire pytest suite to halt at collection-time.  Keeping TVMScript parsing inside the test body causes the failure to occur within the execution of that unit test.

This is a large usability issue when debugging the TVMScript parser, as it becomes impossible to tell how many test failures exist, let alone if there are any commonalities between them.